great_expectations/datasource/fluent/pandas_azure_blob_storage_datasource.py

```diff
@@ -55,11 +59,26 @@ class PandasAzureBlobStorageDatasource(_PandasFilePathDatasource):
     def _get_azure_client(self) -> azure.BlobServiceClient:
         azure_client: Union[azure.BlobServiceClient, None] = self._azure_client
         if not azure_client:
+            # pull in needed config substitutions using the `_config_provider`
+            if not self._config_provider:
+                need_config_subs: set[str] = {
+                    k
+                    for (k, v) in self.azure_options.items()
+                    if isinstance(v, ConfigStr)
+                }
+                if need_config_subs:
+                    warnings.warn(
+                        f"`azure_options` config variables '{','.join(need_config_subs)}' need substitution but no `_ConfigurationProvider` is present"
+                    )
```

Comment:
Going to move this until a utility function so we can re-use this.
Now that is a warning, our tests will catch this as an exception.